core/src/main/java/org/apache/iceberg/rest/responses/ConfigResponse.java

```diff
@@ -178,8 +199,14 @@ public Builder withEndpoints(List<Endpoint> endpointsToAdd) {
       return this;
     }
 
+    /** Sets the optional idempotency key lifetime advertised by the server. */
+    public Builder withIdempotencyKeyLifetime(@Nullable String lifetime) {
+      this.idempotencyKeyLifetime = lifetime;
```

Comment:
should we check lifetime string is an `ISO-8601 duration string` ? and add neg case where lifetime returned from server is not a valid string ?

Reply:
I added a check in the parser and also added a neg test.